src/java.base/share/classes/java/util/jar/JarInputStream.java

```diff
@@ -31,11 +31,53 @@
 import jdk.internal.util.jar.JarIndex;
 
 /**
- * The {@code JarInputStream} class is used to read the contents of
- * a JAR file from any input stream. It extends the class
- * {@code java.util.zip.ZipInputStream} with support for reading
- * an optional {@code Manifest} entry. The {@code Manifest}
- * can be used to store meta-information about the JAR file and its entries.
+ * The {@code JarInputStream} class, which extends {@link ZipInputStream},
+ * is used to read the contents of a JAR file from an input stream.
+ * It provides support for reading an optional {@link JarFile#MANIFEST_NAME Manifest}
+ * entry. The {@code Manifest} can be used to store
+ * meta-information about the JAR file and its entries.
+ *
+ * <h2>Accessing the Manifest</h2>
+ * <p>
+ * The {@link #getManifest} method will return the {@code Manifest} when it is
+ * the first entry in the stream, or {@code META-INF/} is the first entry and
+ * the {@code Manifest} is the second entry within the stream.
```

Comment:
"within the stream" should probably be "in the stream" to be consistent the first part of the sentence.

Reply:
Updated as you suggested